src/EnergyPlus/DaylightingManager.cc

```diff
@@ -6794,58 +6695,55 @@ void DayltgInteriorIllum(EnergyPlusData &state, int &ZoneNum) // Zone number
     GlareFlag = false;
     for (int IL = 1; IL <= NREFPT; ++IL) {
-        if (GLRNDX(IL) > state.dataDaylightingData->ZoneDaylight(ZoneNum).MaxGlareallowed) {
+        if (GLRNDX(IL) > thisDaylightControl.MaxGlareallowed) {
             GlareFlag = true;
             break;
         }
     }
 
-    Array3D<Real64> WDAYIL(
+    Array3D<Real64> WDAYIL(2,
+                           NREFPT,
+                           thisDaylightControl.ShadeDeployOrderExtWins.size()); // Illuminance from window at reference point (second index)
+    //   for shade open/closed (first index), the number of shade deployment groups (third index)
+    Array3D<Real64> WBACLU(
         2,
         NREFPT,
-        state.dataDaylightingData->ZoneDaylight(ZoneNum).ShadeDeployOrderExtWins.size()); // Illuminance from window at reference point (second index)
-    //   for shade open/closed (first index), the number of shade deployment groups (third index)
-    Array3D<Real64> WBACLU(2,
-                           NREFPT,
-                           state.dataDaylightingData->ZoneDaylight(ZoneNum)
-                               .ShadeDeployOrderExtWins.size()); // Background illuminance from window at reference point (second index)
+        thisDaylightControl.ShadeDeployOrderExtWins.size()); // Background illuminance from window at reference point (second index)
     //   for shade open/closed (first index), the number of shade deployment groups (third index)
     Array2D<Real64> RDAYIL(NREFPT,
-                           state.dataDaylightingData->ZoneDaylight(ZoneNum)
-                               .ShadeDeployOrderExtWins.size()); // Illuminance from window at reference point after closing shade
-    Array2D<Real64> RBACLU(NREFPT,
-                           state.dataDaylightingData->ZoneDaylight(ZoneNum)
-                               .ShadeDeployOrderExtWins.size()); // Background illuminance from window at reference point after closing shade
+                           thisDaylightControl.ShadeDeployOrderExtWins.size()); // Illuminance from window at reference point after closing shade
+    Array2D<Real64> RBACLU(
+        NREFPT,
+        thisDaylightControl.ShadeDeployOrderExtWins.size()); // Background illuminance from window at reference point after closing shade
     if (GlareFlag) {
         // Glare is too high at a ref pt.  Loop through windows.
         int count = 0;
 
         continueOuterLoop = false;
-        for (std::size_t igroup = 1; igroup <= state.dataDaylightingData->ZoneDaylight(ZoneNum).ShadeDeployOrderExtWins.size(); igroup++) {
+        for (std::size_t igroup = 1; igroup <= thisDaylightControl.ShadeDeployOrderExtWins.size(); igroup++) {
 
-            std::vector<int> listOfExtWin = state.dataDaylightingData->ZoneDaylight(ZoneNum).ShadeDeployOrderExtWins[igroup - 1];
+            std::vector<int> listOfExtWin = thisDaylightControl.ShadeDeployOrderExtWins[igroup - 1];
```

Comment:
Are you trying to make a copy of the vector here? Or should this be a `const &`?

Reply:
@mjwitte is making a copy appropriate here? (I'm assuming not)

Reply:
`const &listOfExtWin` changed in 3 places.